File: renault2/src/app/Da.ts
import { Client } from './Client';
import { Cdp } from './Cdp';
import { Pilote_Da } from './Pilote_Da';
import { Specialiste } from './Specialiste';
import { Correspondant } from './Correspondant';

export class Da {


  constructor(
    public id?: number,
    public client?: Client,
    public cdp?: Cdp,
    public pilote_da?: Pilote_Da,
    public specialiste?: Specialiste,
    public correspondant?: Correspondant,
    public description_du_besoin?: string,
    public devis_propose?: string,
    public direction_exploitant?: string,
    public service_exploitant?: string,
    public site_utilisation?: string,
    public validation_du_besoin?: boolean,
    public zone_commentaire?: string,
    public version?: number
  ) { }

  // get id(): number {
  //   return this._id;
  // }

  // set id(value: number) {
  //   this._id = value;
  // }

  // get client(): Client {
  //   return this._client;
  // }

  // set client(value: Client) {
  //   this._client = value;
  // }

  // get correspondant(): Correspondant {
  //   return this._correspondant;
  // }

  // set correspondant(value: Correspondant) {
  //   this._correspondant = value;
  // }

  // get specialiste(): Specialiste {
  //   return this._specialiste;
  // }

  // set specialiste(value: Specialiste) {
  //   this._specialiste = value;
  // }

  // get cdp(): Cdp {
  //   return this._cdp;
  // }

  // set cdp(value: Cdp) {
  //   this._cdp = value;
  // }

  // get pilote_da(): Pilote_Da {
  //   return this._pilote_da;
  // }

  // set pilote_da(value: Pilote_Da) {
  //   this._pilote_da = value;
  // }

  // get validation_du_besoin(): boolean {
  //   return this._validation_du_besoin
  // }

  // set validation_du_besoin(value: boolean) {
  //   this._validation_du_besoin = value;
  // }

  // get description_du_besoin(): string {
  //   return this._description_du_besoin
  // }

  // set description_du_besoin(value: string) {
  //   this._description_du_besoin = value;
  // }

  // get devis_propose(): string {
  //   return this._devis_propose
  // }

  // set devis_propose(value: string) {
  //   this._devis_propose = value;
  // }

  // get service_exploitant(): string {
  //   return this._service_exploitant
  // }

  // set service_exploitant(value: string) {
  //   this._service_exploitant = value;
  // }

  // get direction_exploitant(): string {
  //   return this._direction_exploitant
  // }

  // set direction_exploitant(value: string) {
  //   this._direction_exploitant = value;
  // }

  // get site_utilisation(): string {
  //   return this._site_utilisation
  // }

  // set site_utilisation(value: string) {
  //   this._site_utilisation = value;
  // }

  // get zone_commentaire(): string {
  //   return this._zone_commentaire
  // }

  // set zone_commentaire(value: string) {
  //   this._zone_commentaire = value;
  // }

  // get version(): number {
  //   return this._version
  // }

  // set version(value: number) {
  //   this._version = value;
  // }

}

File: projetRenault/src/main/java/com/example/demo/metier/Da.java
package com.example.demo.metier;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;
import javax.persistence.Version;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
@Table(name="DA")
public class Da {

	@Id
	@GeneratedValue (strategy = GenerationType.SEQUENCE, generator = "da_generator")
	@SequenceGenerator (name = "da_generator", sequenceName = "da_seq", allocationSize = 1)
	@JsonView(JsonViews.Common.class)
	private Integer id;
	@Version
	@JsonView(JsonViews.Common.class)
	private Integer version;
	
	@ManyToOne
	@JsonView(JsonViews.ClientAvecDa.class)
	private Client client;
	@ManyToOne
	private Correspondant correspondant;
	@ManyToOne
	private Specialiste specialiste;
	@ManyToOne
	private Cdp cdp;
	@ManyToOne
	private Pilote_Da pilote_da;
	@JsonView(JsonViews.Common.class)
	private String service_exploitant;
	@JsonView(JsonViews.Common.class)
	private String direction_exploitant;
	@JsonView(JsonViews.Common.class)
	private String site_utilisation;
	@JsonView(JsonViews.Common.class)
	private String description_du_besoin;
	@JsonView(JsonViews.Common.class)
	private boolean validation_du_besoin;
	@JsonView(JsonViews.Common.class)
	private String devis_propose;
	@JsonView(JsonViews.Common.class)
	private String zone_commentaire;
	
	
	
	public Da() {
		super();
	}
	
	
	
	public Integer getVersion() {
		return version;
	}



	public void setVersion(Integer version) {
		this.version = version;
	}



	public Integer getId() {
		return id;
	}
	public void setId(Integer id) {
		this.id = id;
	}
	public Client getClient() {
		return client;
	}
	public void setClient(Client client) {
		this.client = client;
	}
	public Correspondant getCorrespondant() {
		return correspondant;
	}
	public void setCorrespondant(Correspondant correspondant) {
		this.correspondant = correspondant;
	}
	public Specialiste getSpecialiste() {
		return specialiste;
	}
	public void setSpecialiste(Specialiste specialiste) {
		this.specialiste = specialiste;
	}
	public Cdp getCdp() {
		return cdp;
	}
	public void setCdp(Cdp cdp) {
		this.cdp = cdp;
	}
	public Pilote_Da getPilote_da() {
		return pilote_da;
	}
	public void setPilote_da(Pilote_Da pilote_da) {
		this.pilote_da = pilote_da;
	}
	public String getService_exploitant() {
		return service_exploitant;
	}
	public void setService_exploitant(String service_exploitant) {
		this.service_exploitant = service_exploitant;
	}
	public String getDirection_exploitant() {
		return direction_exploitant;
	}
	public void setDirection_exploitant(String direction_exploitant) {
		this.direction_exploitant = direction_exploitant;
	}
	public String getSite_utilisation() {
		return site_utilisation;
	}
	public void setSite_utilisation(String site_utilisation) {
		this.site_utilisation = site_utilisation;
	}
	public String getDescription_du_besoin() {
		return description_du_besoin;
	}
	public void setDescription_du_besoin(String description_du_besoin) {
		this.description_du_besoin = description_du_besoin;
	}
	public boolean isValidation_du_besoin() {
		return validation_du_besoin;
	}
	public void setValidation_du_besoin(boolean validation_du_besoin) {
		this.validation_du_besoin = validation_du_besoin;
	}
	public String getDevis_propose() {
		return devis_propose;
	}
	public void setDevis_propose(String devis_propose) {
		this.devis_propose = devis_propose;
	}
	public String getZone_commentaire() {
		return zone_commentaire;
	}
	public void setZone_commentaire(String zone_commentaire) {
		this.zone_commentaire = zone_commentaire;
	}
	
	
	
	
}
